component/src/main/java/io/siddhi/extension/io/cdc/source/metrics/PollingMetrics.java

```diff
@@ -0,0 +1,189 @@
+                                        / 1000;
+                            }
+                            return 0L;
+                        });
+    }
+
+    @Override
+    protected void setCDCDBStatusMetric() {
+        MetricsDataHolder.getInstance().getMetricService()
+                .gauge(String.format("io.siddhi.SiddhiApps.%s.Siddhi.Cdc.Source.Polling.%s.%s",
+                        siddhiAppName, "db_status", getDatabaseURL()),
+                        Level.INFO, () -> {
+                            if (CDC_STATUS_MAP.containsKey(cdcDatabase)) {
+                                return CDC_STATUS_MAP.get(cdcDatabase).ordinal();
+                            }
+                            return -1;
+                        });
+    }
+
+    private void setEventsInLastPollingMetric() {
+        MetricsDataHolder.getInstance().getMetricService()
+                .gauge(String.format("io.siddhi.SiddhiApps.%s.Siddhi.Cdc.Source.Polling.%s.%s",
+                        siddhiAppName, "events_in_last_polling_interval", getDatabaseURL()),
+                        Level.INFO, () -> receiveEventsPerPollingInterval);
+    }
+
+    public void pollingDetailsMetric(int events, long startedTime, long duration, CDCStatus cdcStatus) {
+        String metricName = String.format("io.siddhi.SiddhiApps.%s.Siddhi.Cdc.Source.Polling.Details.%s.%s.%s.%s",
+                siddhiAppName, startedTime, duration, cdcStatus.ordinal(), getDatabaseURL());
+        MetricsDataHolder.getInstance().getMetricService()
+                .gauge(metricName, Level.INFO, () -> events);
+        pollingDetails.add(metricName);
+        if (pollingDetails.size() > pollingHistorySize) {
+            String poll = pollingDetails.poll();
+            MetricsDataHolder.getInstance().getMetricService().remove(poll);
+        }
+    }
+
+    @Override
+    public synchronized void setCDCStatus(CDCStatus cdcStatus) {
+        if (cdcStatus == CDCStatus.ERROR) {
+            getTotalErrorCountMetric().inc();
+        }
+        if (CDC_STATUS_MAP.containsKey(cdcDatabase)) {
+            CDC_STATUS_MAP.replace(cdcDatabase, cdcStatus);
+        } else {
+            CDC_STATUS_MAP.put(cdcDatabase, cdcStatus);
+            setCDCDBStatusMetric();
+        }
+    }
+
+    @Override
+    public synchronized void setLastReceivedTime(long lastPublishedTime) {
+        if (CDC_LAST_RECEIVED_TIME_MAP.containsKey(cdcDatabase)) {
+            if (CDC_LAST_RECEIVED_TIME_MAP.get(cdcDatabase) < lastPublishedTime) {
+                CDC_LAST_RECEIVED_TIME_MAP.replace(cdcDatabase, lastPublishedTime);
+            }
+        } else {
+            CDC_LAST_RECEIVED_TIME_MAP.put(cdcDatabase, lastPublishedTime);
+            lastReceivedTimeMetric();
+            setEventsInLastPollingMetric();
+            idleTimeMetric();
+        }
+    }
+
+    @Override
+    protected String getDatabaseURL() {
+            return host + "/" + tableName;
```

Comment:
use proper indents

Reply:
Fixed in 9b553da8e0ca2f9d10e0868734562af8cfac30e7